.github/workflows/bundle-size.yml

```diff
@@ -0,0 +1,90 @@
+name: Bundle Size Analysis
+
+on:
+  pull_request:
+    paths:
+      - "packages/core/src/components/**"
+      - "packages/core/package.json"
+      - ".size-limit.js"
+
+permissions:
+  pull-requests: write
+
+jobs:
+  build:
+    name: Build
+    uses: ./.github/workflows/build-and-upload.yml
+    secrets:
+      npm_token: ${{ secrets.npm_token }}
+
+  bundle-size:
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - name: Checkout PR
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Download build artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: ci-builds-${{ github.run_id }}-${{ github.run_attempt }}
+          path: packages/
+
+      - name: Run size-limit on PR
+        run: |
+          yarn install --frozen-lockfile
+          node scripts/bundle-check/generate-size-limit-config.js
+          ./node_modules/.bin/size-limit --json > pr.json
+
+      - name: Run size-limit on base commit
+        run: |
+          PR_HEAD_SHA=$(git rev-parse HEAD)
+          trap "echo 'Checking out back to PR commit'; git checkout --force $PR_HEAD_SHA" EXIT
+
+          BASE=$(git merge-base origin/master HEAD)
+          echo "Base commit: $BASE"
+          git checkout $BASE
+
+          if [ -f .size-limit.js ]; then
+            echo ".size-limit.js found on base commit. Running size-limit."
+            yarn install --frozen-lockfile
+            yarn build
+            yarn add --dev -W size-limit @size-limit/preset-small-lib
+            ./node_modules/.bin/size-limit --json > base.json
+          else
+            echo ".size-limit.js not found on base commit. Creating empty base.json."
+            echo "[]" > base.json
+          fi
+        continue-on-error: true
```

Comment:
you can do
```
        env:
          BASE_SHA: ${{ github.event.pull_request.base.sha }}
          HEAD_SHA: ${{ github.sha }}
```

to get the same shas that you're calculating inside, you get them from github's context

also, you can just use the "actions/checkout" instead of doing `git checkout --force $PR_HEAD_SHA" EXIT`
just add a step of "now I'm checkouting to commit $BASE_SHA" like this:
```
- uses: actions/checkout@v4
   env:
    PR_HEAD_SHA: ${{ github.event.pull_request.base.sha }}
   with:
     ref: $PR_HEAD_SHA
```

and then do again the run setup, run build and whatever you need later

Reply:
or just of course straight:
```
- uses: actions/checkout@v4
  with:
    ref: ${{ github.event.pull_request.base.sha }}
```

should only make sure if it is `base.sha` or `head.sha`, I don't remember tbh

Reply:
That's actually seem much nicer, I'll use that, thanks!